### PR TITLE
chore: remove stale snapshots & duplicate tests

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- When reviewing in the cloud, a few tests were marked as modified that had not changes. It appears we have a few duplicate tests.
- The snapshots are stale files that should have been removed in v10 when the runner logic was pulled in packages/app.
- Replace usage of @bahmutov/cly to scaffold a simple example to verify the binary against and use the `cypress-test-tiny` repository instead. Note observed issue: https://github.com/cypress-io/cypress/pull/28738#issuecomment-1896319777

### PR Tasks

- [ ] Have tests been added/updated?
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

File: packages/driver/cypress/e2e/commands/actions/select.cy.js [hunk 1]
```diff
@@ -87,12 +87,6 @@ describe('src/cy/commands/actions/select', () => {
       })
     })
 
-    it('can select an array of values', () => {
```
Comment (emilyrohrbough) [inline packages/driver/cypress/e2e/commands/actions/select.cy.js:90]: duplicate of test on line 62

File: packages/driver/cypress/e2e/cypress/error_utils.cy.ts [hunk 2]
```diff
@@ -372,17 +372,6 @@ describe('driver/src/cypress/error_utils', () => {
     })
   })
 
-  context('.throwErrByPath', () => {
```
Comment (emilyrohrbough) [inline packages/driver/cypress/e2e/cypress/error_utils.cy.ts:375]: duplicate of tests on line 354

File: packages/launchpad/cypress/e2e/migration.cy.ts [hunk 3]
```diff
@@ -637,55 +637,6 @@ describe('Full migration flow for each project', { retries: { openMode: 0, runMo
     checkOutcome()
   })
 
-  it('completes journey for migration-e2e-custom-test-files', () => {
```
Comment (emilyrohrbough) [inline packages/launchpad/cypress/e2e/migration.cy.ts:640]: duplicate of test on (now) line 640

File: packages/driver/cypress/e2e/commands/actions/select.cy.js (hunk 1 above)
Comment (jennifer-shehane) [reply packages/driver/cypress/e2e/commands/actions/select.cy.js:90]: weird

**Review by jennifer-shehane — APPROVED** [review]

**Review by AtofStryker — APPROVED** [review]

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**